benches/common/supernova/mod.rs

```diff
@@ -0,0 +1,166 @@
+// Code is considered dead unless used in all benchmark targets
+#![allow(dead_code)]
+
+pub mod bench;
+pub mod targets;
+
+use anyhow::anyhow;
+use arecibo::{
+  supernova::NonUniformCircuit,
+  supernova::{StepCircuit, TrivialTestCircuit},
+  traits::Engine,
+};
+use bellpepper_core::{num::AllocatedNum, ConstraintSystem, SynthesisError};
+use core::marker::PhantomData;
+use ff::PrimeField;
+
+pub type E1 = arecibo::provider::PallasEngine;
+pub type E2 = arecibo::provider::VestaEngine;
+pub type EE1 = arecibo::provider::ipa_pc::EvaluationEngine<E1>;
+pub type EE2 = arecibo::provider::ipa_pc::EvaluationEngine<E2>;
+// SNARKs without computation commitments
+pub type S1 = arecibo::spartan::batched::BatchedRelaxedR1CSSNARK<E1, EE1>;
+pub type S2 = arecibo::spartan::snark::RelaxedR1CSSNARK<E2, EE2>;
+// SNARKs with computation commitments
+pub type SS1 = arecibo::spartan::batched_ppsnark::BatchedRelaxedR1CSSNARK<E1, EE1>;
+pub type SS2 = arecibo::spartan::ppsnark::RelaxedR1CSSNARK<E2, EE2>;
+
+// This should match the value in test_supernova_recursive_circuit_pasta
+// TODO: This should also be a table matching the num_augmented_circuits in the below
+pub const NUM_CONS_VERIFIER_CIRCUIT_PRIMARY: usize = 9844;
+pub const NUM_SAMPLES: usize = 10;
```

Comment:
Not-so-fun part: this number is meant to be meaningfully different for the nova and supernova use cases. We may want to document that.
This number is also a footgun waiting to happen, see my note on `let num_cons = ...`